projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/facet-list/facet-list.component.ts

```diff
@@ -57,19 +70,27 @@ export class FacetListComponent {
     trap: true,
     block: true,
     focusOnEscape: true,
-    autofocus: 'cx-facet',
+    autofocus: 'cx-facet > button',
   };
 
   @HostListener('click') handleClick() {
     this.close();
   }
+  @Optional() focusService = inject(KeyboardFocusService, { optional: true });
+  @Optional() featureConfigService = inject(FeatureConfigService, {
+    optional: true,
+  });
 
   constructor(
     protected facetService: FacetService,
     protected elementRef: ElementRef,
     protected renderer: Renderer2
   ) {}
 
+  ngOnInit(): void {
+    this.subscribeToFocusUpdates();
```

Comment:
Lets move the feature flag here instead to make our method more simple.

```suggestion
    if(this.featureConfigService?.isEnabled('a11yFacetsDialogFocusHandling')) {
        this.subscribeToFocusUpdates();
    }
```